ts/components/session/SessionRadio.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+interface Props {
+  label: string;
+  value: string;
+  active: boolean;
+  group?: string;
+  onClick: any;
+}
+
+interface State {
+  active: boolean;
+}
+
+export class SessionRadio extends React.PureComponent<Props, State> {
+  public static defaultProps = {
+    onClick: () => null,
+  };
+
+  constructor(props: any) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+
+    this.state = {
+      active: this.props.active,
+    };
+  }
+
+  public render() {
+    const active = this.state.active;
+    const { label, group, value } = this.props;
+
+    return (
+      <div className="session-radio">
+        <input
+          type="radio"
+          name={group || ''}
+          value={value}
+          defaultChecked={active}
+          aria-checked={active}
+          onClick={this.clickHandler}
+        />
+        <label>{label} </label>
+      </div>
+    );
+  }
+
+  private clickHandler(e: any) {
+    if (this.props.onClick) {
+      e.stopPropagation();
+      this.props.onClick();
+
+      this.setState({
+        active: !this.state.active,
```

Comment:
in React you should never access `this.state` in the setState as setState is asynchronous and calls can be batched, you have to  use this form instead:
```
setState(prevState =>{
   return{
        ...prevState,
        counter : prevState.counter +1
   }
})

```